### PR TITLE
make accessing query result records in ascii view type safe

### PR DESCRIPTION
**Pull request by HerrEmil**
A crash was reported when closing a frame while it is running a query, with a screenshot that showed a runtime type error when evaluating `result.records.length` when result was null. I was unable to recreate exactly the error in the screenshot, but could trigger a runtime type error when evaluating `result.records.length` in ascii view by opening the text tab, re-running a query and closing the frame while the query was running.

This PR fixes the potential type error in ascii view and also adds stricter typings to the ascii view component, requests duck file and other files connected to these files to make it easier to fix any other unknown type errors when evaluating query result in other components.

### CODE REVIEW

**Review by OskarDamkjaer — COMMENTED** [review]
Great to see these kinds of changes 💯

File: src/browser/modules/Stream/CypherFrame/AsciiView.tsx [hunk 1]
```diff
@@ -121,29 +142,44 @@ export class AsciiViewComponent extends Component<
...
 
-type AsciiStatusbarComponentState = any
+interface AsciiStatusbarComponentProps {
+  _asciiMaxColWidth?: number
+  result: BrowserRequestResult
+  maxRows: number
+  maxFieldItems: number
+  updated?: number
+  _asciiSetColWidth?: number
+  setParentState: { (state: any): void }
```
[the remark this answers is not in the document]
Comment (HerrEmil) [reply src/browser/modules/Stream/CypherFrame/AsciiView.tsx]: I removed the destructured parent state and the passing down of the setState method to Ascii view now, it was only passing and updating two values. I did not update the other cypher views.

File: src/browser/modules/Stream/CypherScriptFrame/Summary.tsx [hunk 2]
```diff
@@ -156,7 +157,8 @@ export const Summary = ({
         <PaddedStatsBar>
           <StyledCypherErrorMessage>ERROR</StyledCypherErrorMessage>
           <MessageArea>
-            {(request.result || {}).message || 'Unknown error'}
+            {((request.result || {}) as BrowserError).message ||
```
[the remark this answers is not in the document]
Comment (HerrEmil) [reply src/browser/modules/Stream/CypherScriptFrame/Summary.tsx]: Agreed, changed it

File: src/browser/modules/Stream/Stream.tsx [hunk 3]
```diff
@@ -124,7 +128,7 @@ function Stream(props: StreamProps): JSX.Element {
           stack: frameObject.stack
         }
 
-        let MyFrame = getFrame(frame.type)
+        let MyFrame = getFrame(frame.type as FrameType)
```
[the remark this answers is not in the document]
Comment (HerrEmil) [reply src/browser/modules/Stream/Stream.tsx:131]: Yes, I tried but it was not trivial to share the type unfortunately

File: src/shared/modules/commands/helpers/cypher.ts [hunk 4]
```diff
@@ -46,6 +46,6 @@ export const handleCypherCommand = (
       useDb: action.useDb
     }
   )
-  put(send('cypher', id))
+  put(send(id))
```
[the remark this answers is not in the document]
Comment (HerrEmil) [reply src/shared/modules/commands/helpers/cypher.ts:49]: The function was always called with the same value for the first parameter so I inlined the value

**Review by OskarDamkjaer — APPROVED** [review]
Nice changes, merge on green!

**Pull request merged**